File: Assets/Scripts/Player/Reference Scripts/Movement.js
#pragma strict

var player : Transform;
var conversation : boolean;
var animator : Animator;

var speed : float;
var movementVector : Vector3;

var horizontalObject : GameObject;
var verticalObject : GameObject;
var sensitivity : float;
var lock : boolean;

var boundaryHit : RaycastHit;
var raycastVector : Vector3;
var boundaryDistance : float;
var raycastPoint :Transform;

var jumpHit : RaycastHit;
var jumpHeight : float;
var jumpRayDistance : float;

function Start () {
	lock = true;
}
//Als de speler in gesprek is zal in de update ook beweging worden uitgezet tot dat hij weer uitgesprek is
function Update () {
    if(conversation == false){
        Movement ();
        Jump ();
        Boundaries();
		CameraMovement();
    }
}
// Zorgt voor de beweging van de player.
function Movement () {
    movementVector = Vector3(Input.GetAxis("Horizontal"),0,Input.GetAxis("Vertical"));
}
function CameraMovement (){
	verticalObject.transform.Rotate(Vector3(-Input.GetAxis("Mouse Y"),0,0) * sensitivity * Time.deltaTime );
	horizontalObject.transform.Rotate(Vector3(0,Input.GetAxis("Mouse X"),0) * sensitivity * Time.deltaTime );
	if(lock == true){
		Cursor.visible = false;
		Cursor.lockState = Cursor.lockState.Locked;
	}
	else{
		Cursor.visible = true;
		Cursor.lockState = Cursor.lockState.None;
	}
}
// deze functie laat de speler springen als er grond onder hem is.
function Jump () {
	if(Input.GetButtonDown("Jump")){
		if(Physics.Raycast(player.position,player.position.down, jumpHit, jumpRayDistance)){
		    if(jumpHit.transform.tag == "Terrain"){    
		        player.GetComponent.<Rigidbody>().velocity.y = jumpHeight;
		    }
		}
	}
} 
// Dit is het systeem dat ervoor zorgt dat de speler niet door muren heen glitched. 
// Het zal een raycast schieten naar de kant die hij wilt op lopen en kijken of er iets is waar hij tegen aan zou lopen. Als er niks is kan hij gewoon lopen.
function Boundaries () {
    var rayColor : Color ;
    raycastPoint.Translate(movementVector);
    if(movementVector == Vector3.zero){
        raycastPoint.position = player.position;
    }
    if(Physics.Raycast(player.transform.position,raycastPoint.position, boundaryHit,boundaryDistance)){
        print(boundaryHit.transform.name); 
        rayColor = Color.red;
        
        
    }
    else{
        player.Translate(movementVector * speed * Time.deltaTime);
        rayColor = Color.green;
    }
    Debug.DrawRay(player.position,raycastPoint.position * 100,rayColor);
}
//Activeert een loop animatie wanneer de speler begint met lopen.
function Animations () {
	if(movementVector == Vector3.zero){
		animator.SetBool("Walking", false);
	}
	else{
		animator.SetBool("Walking", true);
	}
}

File: Assets/Scripts/Player/Movement.cs
using UnityEngine;
using System.Collections;

public class Movement : MonoBehaviour {
	public Transform playerObj;
	public Transform player;
	public Animator anim;
	
	public GameObject vObj;
	public GameObject hObj;
	public float sensitivity;
	public bool lockS;
	
	public float speed;
	
	public float jumpPower;
	public float jumpRayD;
	public Rigidbody playerP;
	
	public bool camMove = true;
	public bool move = true;
	public bool jump = true;

	public Vector3 fallRespawn;
	public Transform teleporter;

	public float updownRange = 90.0f;
 	float verticalRotation = 0;
 	float horizontalRotation = 0;
	public float mouseSensitivity = 4.0f;

	
	void Start () {
		lockS = true;
		Mouse();
	}
	void Update () {
		if(camMove == true){
			CameraMoving();
		}
		if(move == true){
			Moving();
		}
		if(jump == true){
			Jumping();
		}
	}
	void CameraMoving () {
		horizontalRotation -= -Input.GetAxis ("Mouse X") * mouseSensitivity;
		hObj.transform.localRotation = Quaternion.Euler (0, horizontalRotation, 0 );

        verticalRotation -= -Input.GetAxis ("Mouse Y") * mouseSensitivity;
     	verticalRotation = Mathf.Clamp (verticalRotation, -updownRange, updownRange);
        vObj.transform.localRotation = Quaternion.Euler (verticalRotation, 0, 0);

	}
	void Moving () {
		Vector3 mvr = new Vector3(Input.GetAxis("Horizontal"),0,Input.GetAxis("Vertical"));
		playerObj.Translate(mvr * speed * Time.deltaTime);
		if(mvr == Vector3.zero){
			anim.SetBool("Running",false);
		}
		else{
			anim.SetBool("Running",true);
		}
	}
	void Jumping () {
		Vector3 jvr = transform.TransformDirection(-Vector3.up * 10);
		if(Input.GetButtonDown("Jump")){
			if(teleporter != null) {
				teleporter.transform.GetComponent<Teleporter>().Teleport(playerObj );			}
			else{
				if(Physics.Raycast(player.position,jvr,jumpRayD)){
					playerP.AddForce(transform.up*jumpPower);
					print("Jump Accomplished");
				}
			}		
		}
	}
	public void ToggleMovement () {
		camMove = !camMove;
		jump = !jump;
		move = !move;
		lockS = !lockS;
		Mouse();
	}
	public void Mouse () {
		if(lockS == true){
			Cursor.visible = false;
			Cursor.lockState = CursorLockMode.Locked;
		}
		else{
			Cursor.visible = true;
			Cursor.lockState = CursorLockMode.None;
		}
	}
}
